test/integration/controller/jobs/pod/pod_controller_test.go

```diff
@@ -1400,3 +1401,196 @@ var _ = ginkgo.Describe("Pod controller interacting with scheduler", ginkgo.Orde
 		})
 	})
 })
+
+var _ = ginkgo.Describe("Pod controller interacting with Workload controller when waitForPodsReady enabled", ginkgo.Ordered, ginkgo.ContinueOnFailure, func() {
+	var (
+		ns *corev1.Namespace
+		fl *kueue.ResourceFlavor
+		cq *kueue.ClusterQueue
+		lq *kueue.LocalQueue
+	)
+
+	ginkgo.BeforeAll(func() {
+		fwk = &framework.Framework{
+			CRDPath:     crdPath,
+			WebhookPath: webhookPath,
+		}
+		cfg = fwk.Init()
+		waitForPodsReady := &configapi.WaitForPodsReady{
+			Enable:  true,
+			Timeout: &metav1.Duration{Duration: 10 * time.Millisecond},
+			RequeuingStrategy: &configapi.RequeuingStrategy{
+				Timestamp:          ptr.To(configapi.EvictionTimestamp),
+				BackoffLimitCount:  ptr.To[int32](1),
+				BackoffBaseSeconds: ptr.To[int32](1),
+			},
+		}
+		ctx, k8sClient = fwk.RunManager(cfg, managerSetup(
+			&configapi.Configuration{WaitForPodsReady: waitForPodsReady},
```

Comment:
Is this necessary? We pass this information already in `jobframework.WithWaitForPodsReady(waitForPodsReady),`. Looking at some other tests looks like this is enough, for example, [here](https://github.com/kubernetes-sigs/kueue/blob/8c182be3290b6ec372921762d56c500ee123d5cd/test/integration/controller/jobs/job/job_controller_test.go#L841)

Reply:
Yes, because this is configuration for manager and workload controller. But `jobframework.WithWaitForPodsReady(waitForPodsReady)` is for jobcontrollers.

Reply:
https://github.com/kubernetes-sigs/kueue/blob/194dc8eec422fa1fe8075b8c90129ea06b2199bd/test/integration/controller/jobs/pod/suite_test.go#L93-L96

Reply:
In [this](https://github.com/kubernetes-sigs/kueue/blob/8c182be3290b6ec372921762d56c500ee123d5cd/test/integration/controller/jobs/job/job_controller_test.go#L841) case not using workload_controller

Reply:
Ah, got it! Thanks for clarifying. In that case can we append `jobframework.WithWaitForPodsReady(waitForPodsReady)` within `managerSetup`? So that we don't need to pass the same information in two ways.

Reply:
Done